app/models/v2/report.rb

```diff
@@ -7,9 +7,76 @@ class Report < ApplicationRecord
     self.table_name = :reports
     self.primary_key = :id
 
+    SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(Pundit.policy_scope(User.current, V2::System).where_clause.ast)
+    end
+
+    COMPLIANT_SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(
+        Pundit.policy_scope(User.current, V2::System).where_clause.ast.and(
+          V2::TestResult.arel_table[:score].gteq(V2::Report.arel_table[:compliance_threshold])
+        )
+      )
```

Comment:
Can be applied to all of the constants:
```suggestion
      AN::NamedFunction.new('COUNT', [V2::System.arel_table[:id]]).filter(
        Pundit.policy_scope(User.current, V2::System).where_clause.ast.and(
          V2::TestResult.arel_table[:score].gteq(V2::Report.arel_table[:compliance_threshold])
        )
      )
```